src/ray/core_worker/core_worker.cc

```diff
@@ -1168,15 +1290,16 @@ Status CoreWorker::GetOwnerAddress(const ObjectID &object_id,
 std::vector<rpc::ObjectReference> CoreWorker::GetObjectRefs(
     const std::vector<ObjectID> &object_ids) const {
   std::vector<rpc::ObjectReference> refs;
+  refs.reserve(object_ids.size());
   for (const auto &object_id : object_ids) {
     rpc::ObjectReference ref;
     ref.set_object_id(object_id.Binary());
     rpc::Address owner_address;
     if (reference_counter_->GetOwner(object_id, &owner_address)) {
       // NOTE(swang): Detached actors do not have an owner address set.
-      ref.mutable_owner_address()->CopyFrom(owner_address);
+      *ref.mutable_owner_address() = std::move(owner_address);
     }
-    refs.push_back(std::move(ref));
+    refs.emplace_back(std::move(ref));
```

Comment:
i don't think emplace has any perf benefit over push in a situation like this where they're the same type, just generates more templated code, and ends up doing the same thing with one rvalue forward and one move construction

Reply:
> i don't think emplace has any perf benefit over push in a situation like this where they're the same type

Push back = default construction + move assignment
Emplace back = in-place construction

Reply:
push takes the vector type as parameter and forwards to the placement new
emplace takes args to construct and fowards to the placement new, so would save a temporary extra being created from the args as the parameter that would happen in the push case if you pass in args to construct

looks something like this iiuc
```
push(T&& elem) { 
    new (backing+size) T(forward(elem)) 
}
```
```
emplace(Args&&... args) { 
    new (backing+size) T(forward(args)...);
 }
```
So when you're calling with an object with the type of the vector they're the same thing -  take as ref and then construct, just emplace has extra compiler overhead of variadic template stuff, super trivial thing either way though